src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -165,11 +165,35 @@ osexperimental_handle_live_fs (RPMOSTreeOSExperimental *interface,
   return TRUE;
 }
 
+static gboolean
+osexperimental_handle_search(RPMOSTreeOSExperimental *interface,
+                             GDBusMethodInvocation *invocation,
+                             const char *query)
+{
+  g_autoptr(GCancellable) cancellable = g_cancellable_new ();
+  GError **error = NULL;
+  OstreeSysroot *sysroot = rpmostreed_sysroot_get_root (rpmostreed_sysroot_get ());
+  OstreeRepo *repo = ostree_sysroot_repo (sysroot);
+  g_autoptr(RpmOstreeContext) ctx = rpmostree_context_new_client (repo);
+  rpmostree_context_set_dnf_caching (ctx, RPMOSTREE_CONTEXT_DNF_CACHE_FOREVER);
+
+  if (!rpmostree_context_download_metadata (ctx, DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_RPMDB,
+                                            cancellable, error))
+    return FALSE;
+
+  DnfSack *sack = dnf_context_get_sack (ctx);
+  g_autoptr(GPtrArray) urls = rpmostree_get_matching_packages(sack, query);
```

Comment:
This function returns an array of `DnfPackage`s, not URLs. It *usually* returns a single package, but e.g. if a user queries for "kernel", and the same latest kernel is available from multiple repos, it'll return all of them. Usually, we just pick the first one (see other places where this function is used).

So here, we'll want to do something like (pseudo-code):

```
for each query in queries:
  pkgs = rpmostree_get_matching_packages(sack, query)
  if pkgs list is empty:
    error out
  add dnf_package_get_location(pkgs[0]) to list of urls
return list of urls
```